builtin/providers/rundeck/resource_private_key.go

```diff
@@ -0,0 +1,132 @@
+package rundeck
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/apparentlymart/go-rundeck-api/rundeck"
+)
+
+func resourceRundeckPrivateKey() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateOrUpdatePrivateKey,
+		Update: CreateOrUpdatePrivateKey,
+		Delete: DeletePrivateKey,
+		Exists: PrivateKeyExists,
+		Read:   ReadPrivateKey,
+
+		Schema: map[string]*schema.Schema{
+			"path": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to the key within the key store",
+				ForceNew:    true,
+			},
+
+			"key_material_file": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to a file containing the private key material to store",
+				// Although the user provides the file, we actually
+				// persist a hash of the file's contents in the state so
+				// we can detect if it changes without disclosing the
+				// private key itself into the state.
+				StateFunc: PrivateKeyMaterialStateFunc,
+			},
```

Comment:
We're learning the lesson that it's better to define these attributes such that they accept the contents of a file instead of a path, and let users pull in a path if they need to via the `file()` helper.

So I'd just shift this to `"key_material"`, which should save you a few lines of code as well.